File: anchors/src/main/java/com/effective/android/anchors/LockableTask.java
package com.effective.android.anchors;

class LockableTask extends Task {

    private LockableAnchor lockableAnchor;

    LockableTask(Task wait,LockableAnchor lockableAnchor) {
        super(wait.getId() + "_waiter",true);
        lockableAnchor.setTargetTaskId(wait.getId());
        this.lockableAnchor = lockableAnchor;
    }

    @Override
    protected void run(String name) {
        lockableAnchor.lock();
    }

    boolean successToUnlock(){
        return lockableAnchor.successToUnlock();
    }
}

File: anchors/src/main/java/com/effective/android/anchors/LockableTask.kt
package com.effective.android.anchors

internal class LockableTask(wait: Task, lockableAnchor: LockableAnchor) : Task(wait.id + "_waiter", true) {
    private val lockableAnchor: LockableAnchor
    override fun run(name: String) {
        lockableAnchor.lock()
    }

    fun successToUnlock(): Boolean {
        return lockableAnchor.successToUnlock()
    }

    init {
        lockableAnchor.setTargetTaskId(wait.id)
        this.lockableAnchor = lockableAnchor
    }
}